themes/default/content/docs/languages-sdks/python/python-blocking-async.md

```diff
@@ -0,0 +1,199 @@
+---
+title_tag: "Blocking and Async Python with Pulumi | Languages & SDKs"
+meta_desc: An overview of how to use Python with Pulumi for infrastructure as code on any cloud (AWS, Azure, Google Cloud, Kubernetes, etc.).
+title: Blocking and Async
+h1: Blocking and Async Python with Pulumi
+meta_image: /images/docs/meta-images/docs-meta.png
+menu:
+  languages:
+    parent: python
+    weight: 5
+aliases:
+- /python/async/
+---
+
+In large and complex Python Pulumi programs, blocking calls can prevent progress
+from being made by the Pulumi runtime by preventing the event loop from
+processing asynchronous tasks.  
+
+If the output from the blocking code will be used as an input to a resource,
+then the blocking code may be executed on another thread with the future result
+captured as a `pulumi.Output` object. In the same way, asynchronous code (using
+Python’s [`asyncio`](https://docs.python.org/3/library/asyncio.html)) can also
+be evaluated concurrently with the Pulumi runtime and the future result also
+captured using `pulumi.Output`.
```

Comment:
This graf seems like it follows logically from the rest of the document, but read on its own applies that this is somehow done for the user. Our are often written as though the Pulumi engine or runtime is the subject, so someone could read "the blocking code may be executed on another thread" as though the Pulumi runtime is doing that for them or there's some heuristic involved. 

Perhaps "Blocking could _should_ be executed on another thread"?

Or using a more active voice:

> The Pulumi runtime enables you to await synchronous and asnychronous calls without blocking the event loop. ... [describe that for synchronous calls, we use `asyncio.to_thread` to make synchronous calls async] [describe that for both, we then use `pulumi.Output.from_input`] ...

Reply:
Thanks! I reworded this